memcheck/tests/atomic_incs.c

```diff
@@ -246,8 +246,25 @@ __attribute__((noinline)) void atomic_add_8bit ( char* p, int n )
    } while (block[2] != 1);
 #endif
 #elif defined(VGA_riscv64)
-   /* TODO Implement. */
-   assert(0);
+   unsigned long long int block[3]
+      = { (unsigned long long int)p, (unsigned long long int)n,
+          0xFFFFFFFFFFFFFFFFULL};
+   do {
+      __asm__ __volatile__(
+         "mv     t0, %0"         "\n\t"
+         "ld     t1, (t0)"       "\n\t" // p
+         "ld     t2, 8(t0)"      "\n\t" // n
+         "lr.w   t3, (t1)"       "\n\t"
+         "slli   t3, t3, 56"     "\n\t" // signed-extend
+         "sra    t3, t3, 56"     "\n\t"
+         "add    t3, t3, t2"     "\n\t"
+         "sc.w   t4, t3, (t1)"   "\n\t"
+         "sd     t4, 16(t0)"     "\n\t"
```

Comment:
Atomic add for <32-bit memory locations is a bit problematic because RV64A doesn't have respective instructions. The code could go through hoops to simulate them using `lr.w`/`sc.w` for all cases. For instance, the MIPS variant makes sure that extra bytes that need to be loaded/stored always stay intact. However,  this simpler RISC-V implementation should be ok too, it matches what is done for PPC and looks to be sufficient for what the test actually tries to achieve.